apps/meteor/client/views/modal/uikit/ModalBlock.tsx

```diff
@@ -34,79 +36,91 @@ const focusableElementsStringInvalid = `
 	[tabindex]:not([tabindex="-1"]):invalid,
 	[contenteditable]:invalid`;
 
-function ModalBlock({ view, errors, appId, onSubmit, onClose, onCancel }) {
+type ModalBlockParams = {
+	view: IUIKitSurface & { showIcon?: boolean };
+	errors: any;
+	appId: string;
+	onSubmit: FormEventHandler<HTMLElement>;
+	onClose: () => void;
+	onCancel: FormEventHandler<HTMLElement>;
+};
+
+const KeyboardCode = new Map<string, number>([
+	['ENTER', 13],
+	['ESC', 27],
+	['TAB', 9],
+]);
+
+const ModalBlock = ({ view, errors, appId, onSubmit, onClose, onCancel }: ModalBlockParams): ReactElement => {
 	const id = `modal_id_${useUniqueId()}`;
-	const ref = useRef();
+	const ref = useRef<HTMLElement>(null);
 
-	// Auto focus
 	useEffect(() => {
 		if (!ref.current) {
 			return;
 		}
 
 		if (errors && Object.keys(errors).length) {
-			const element = ref.current.querySelector(focusableElementsStringInvalid);
-			element && element.focus();
+			const element = ref.current.querySelector(focusableElementsStringInvalid) as HTMLElement;
+			element?.focus();
 		} else {
-			const element = ref.current.querySelector(focusableElementsString);
-			element && element.focus();
+			const element = ref.current.querySelector(focusableElementsString) as HTMLElement;
+			element?.focus();
 		}
 	}, [errors]);
-	// save focus to restore after close
-	const previousFocus = useMemo(() => document.activeElement, []);
-	// restore the focus after the component unmount
-	useEffect(() => () => previousFocus && previousFocus.focus(), [previousFocus]);
-	// Handle Tab, Shift + Tab, Enter and Escape
+
+	const previousFocus = useMemo<HTMLElement>(() => document.activeElement as HTMLElement, []);
```

Comment:
`document.activeElement` might be a `SVGElement`, which has no `focus()` method, or `null`. Better keep it as it is (`Element | null`) and use a type guard to ensure that is not `null` and has a `focus()` method.